back/app/services/projects_finder_service.rb

```diff
@@ -4,6 +4,76 @@ def initialize(projects, user = nil, params = {})
     @user = user
     @page_size = (params.dig(:page, :size) || 500).to_i
     @page_number = (params.dig(:page, :number) || 1).to_i
+    @finished = params[:finished]
+    @archived = params[:archived]
+  end
+
+  # Returns an ActiveRecord collection of published projects that are
+  # in an active participatory phase (where user can probably do something),
+  # ordered by the end date of the current phase, soonest first (nulls last).
+  # Also returns action descriptors for each project, to avoid getting them again when serializing.
+  # => { projects: [Project], descriptor_pairs: { <project.id>: { <action_descriptors> }, ... } }
+  def participation_possible
```

Comment:
This is unchanged, other than the name of the method. It's just moved out of the `private` section, as we are no longer calling it from the caching function (caching removed)